src/crypto/qat/qcccrypto.h

```diff
@@ -18,27 +30,41 @@ extern "C" {
 }
 
 class QccCrypto {
+    friend class QatCrypto;
+    size_t chunk_size;
+    size_t max_requests;
+
+    boost::asio::io_context my_context;
+    std::thread qat_context_thread;
+    void my_context_run();
+    using work_guard_type = boost::asio::executor_work_guard<boost::asio::io_context::executor_type>;
+    std::unique_ptr<work_guard_type> work_guard;
```

Comment:
> it looks like all of these members could be replaced with a [`boost::asio::thread_pool`](https://www.boost.org/doc/libs/1_79_0/doc/html/boost_asio/reference/thread_pool.html) constructed with `num_threads=1`. the work guard isn't necessary, because the pool will continue running until `join()` is called

Here work guard is not used to protect thread, but to protect `my_context`, in case `my_context` ends directly without tasks.

Reply:
understood, we use that same work guard pattern with `io_context` in the beast frontend. but if you use `thread_pool` instead of `io_context` it won't stop early until you call [`join()`](https://www.boost.org/doc/libs/1_79_0/doc/html/boost_asio/reference/thread_pool/join.html)

Reply:
I have replaced `io_context` with `thread_pool` and deleted related code. This makes the code much more concise. Thanks.